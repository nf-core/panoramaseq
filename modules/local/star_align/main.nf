process STAR_ALIGN_LOCAL {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/../nf-core/star/genomegenerate/environment.yml"
    container "community.wave.seqera.io/library/htslib_samtools_star_gawk:ae438e9a604351a4"

    input:
    tuple val(meta), path(fastq2), path(genome_dir)

    output:
        tuple val(meta), path("*.bam"), emit: bam
        tuple val(meta), path("*.bai"), optional: true, emit: bai
        tuple val(meta), path("*.cram"), optional: true, emit: cram
        tuple val(meta), path("*.crai"), optional: true, emit: crai
        tuple val(meta), path("*.csi"), optional: true, emit: csi
        path "*_Log.final.out", emit: log_final
        path "*_Log.out", emit: log_out
        path "*_Log.progress.out", emit: log_progress
        path  "versions.yml", emit: versions

    publishDir "${params.outdir}/star", mode: 'copy', overwrite: true

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    STAR \\
        --runThreadN ${task.cpus} \\
        --readFilesCommand zcat \\
        --genomeDir $genome_dir \\
        --readFilesIn $fastq2 \\
        --outFileNamePrefix ${prefix}_ \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        star: \$(STAR --version | head -n 1 | sed 's/STAR_//')
END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_Aligned.sortedByCoord.out.bam
    touch ${prefix}_Log.final.out
    touch ${prefix}_Log.out
    touch ${prefix}_Log.progress.out

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        star: \$(STAR --version | head -n 1 | sed 's/STAR_//')
END_VERSIONS
    """
}
