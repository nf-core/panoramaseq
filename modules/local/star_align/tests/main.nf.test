nextflow_process {

    name "Test Process STAR_ALIGN_LOCAL"
    script "../main.nf"
    process "STAR_ALIGN_LOCAL"

    tag "modules"
    tag "modules_local"
    tag "star_align"

    test("star_align_local - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'S09' ], // meta map
                    file(params.panoramaseq_testdata_base_path + 'star_align/S09.cutadapt.adv.trim2.R2.fastq.gz', checkIfExists: true),
                    file('mock_star_index')  // Mock directory for stub test
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.bam },
                { assert process.out.log_final },
                { assert process.out.log_out },
                { assert process.out.versions },
                { assert snapshot(process.out).match("stub") }
            )
        }
    }
}
