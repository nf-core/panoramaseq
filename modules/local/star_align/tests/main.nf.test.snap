{
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "S09"
                        },
                        "S09.star_Aligned.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    "S09.star_Log.final.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "6": [
                    "S09.star_Log.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "7": [
                    "S09.star_Log.progress.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "8": [
                    "versions.yml:md5,17e5956c5d53aa9485212a66adb000fe"
                ],
                "bai": [
                    
                ],
                "bam": [
                    [
                        {
                            "id": "S09"
                        },
                        "S09.star_Aligned.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "crai": [
                    
                ],
                "cram": [
                    
                ],
                "csi": [
                    
                ],
                "log_final": [
                    "S09.star_Log.final.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "log_out": [
                    "S09.star_Log.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "log_progress": [
                    "S09.star_Log.progress.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,17e5956c5d53aa9485212a66adb000fe"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-06-24T13:19:23.538032405"
    },
    "star_align_local - with genome generation": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "8": [
                    
                ],
                "bai": [
                    
                ],
                "bam": [
                    
                ],
                "crai": [
                    
                ],
                "cram": [
                    
                ],
                "csi": [
                    
                ],
                "log_final": [
                    
                ],
                "log_out": [
                    
                ],
                "log_progress": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-06-24T13:20:30.596820879"
    }
}